### ISSUE TITLE
How does fastText calculate the probabilities reported by predict-prob?

### ISSUE THREAD

**Issue by adamcrussell** [facebookresearch/fastText#405]
In a binary classification problem with two labels (__label__cat and __label__other) when you run the predict-prob command you get some output that looks like: 
    __label__cat 0.625
    __label__other 0.544922
    __label__cat 0.634766
    ...
My understanding is that these are the most likely labels with the associated probability given. 

My questions go like this this: the probabilities sum to 1 and so for the first test record we can say that p(__label__other)=0.375, right? The fact that they sum to 1 like this leads me to assume that these values are simply the result of a typical softmax function, is that correct?

The reason I am asking this seemingly basic question is that I am getting some unexpected results when taking these reported probabilities and using them with scikit-learn metrics functions. For example, if I run a model trained against a set training data set but then run it against total noise (gibberish) that has been assigned real labels (50/50 0/1 labels) I end up with a sklearn.metrics.roc_auc_score() of 0.81, which is very odd. I would expect that in this case that I would get about 0.5, which some minor variation based on random chance.

**Comment by matanster** [comment]
It's softmax indeed, you can see that in the article and the code alike.

**Comment by Celebio** [comment]
Hi @adamcrussell ,
By default the output is a softmax. You can also use the loss "one-vs-all" as [described here](https://fasttext.cc/docs/en/supervised-tutorial.html#multi-label-classification) in order to have independent classifiers for each label.

I am closing the issue now. Feel free to re-open it if you need any further information.

Best regards,
Onur
